### PR TITLE
[OTX/XAI] OTX XAI integration for class-wise, det and cls

### PR DESCRIPTION
**Pull request by dongkwan-kim01**
**Summary**
1. Support explain mode for detection & classification
2. Integration of class-wise saliency map for cls, det

**This PR includes**
Add `DetectionExplainer` class to mpa/det/explainer.py
Support ReciproCAM and class-wise saliency map to detection case
Add explain stage

**Test Codes**
Can test with OTX CLI(please refer OTX cli support, [link](https://github.com/openvinotoolkit/training_extensions/pull/1342))

### CODE REVIEW

**Comment by dongkwan-kim01** [comment]
@goodsong81 @negvet @harimkang Hello, because of the termination date, could you review this PR?

File: mpa/det/inferrer.py [hunk 1]
```diff
@@ -190,7 +186,24 @@ def dummy_dump_saliency_hook(model, input, out):
...
         if eval:
-            metric = dataset.evaluate(eval_predictions, **cfg.evaluation)[cfg.evaluation.metric]
+            metric = dataset.evaluate(eval_predictions, **cfg.evaluation)
+            metric = metric['mAP'] if isinstance(cfg.evaluation.metric, list) else metric[cfg.evaluation.metric]
+
+        # Check and unwrap ImageTilingDataset object from TaskAdaptEvalDataset
+        while hasattr(dataset, 'dataset') and not isinstance(dataset, ImageTilingDataset):
+            dataset = dataset.dataset
+
+        if isinstance(dataset, ImageTilingDataset):
```
Comment (negvet) [inline mpa/det/inferrer.py:183]: Supporting `ImageTilingDataset` on the way.. It is better to do it separately, but not sure that we can afford it in terms of time. I would approve to keep it here :)

**Review by negvet — APPROVED** [review]
Thank you for the good work! LGTM

File: mpa/cls/inferrer.py [hunk 2]
```diff
@@ -110,10 +110,9 @@ def _infer(self, cfg, dump_features=False, dump_saliency_map=False):
                     saliency_maps = forward_explainer_hook.records if dump_saliency_map else [None] * len(self.dataset)
 
         assert len(eval_predictions) == len(feature_vectors) == len(saliency_maps), \
-            (
-                "Number of elements should be the same, however, number of outputs are ",
-                f"{len(eval_predictions)}, {len(feature_vectors)}, and {len(saliency_maps)}"
-            )
+               'Number of elements should be the same, however, number of outputs are ' \
```
Comment (goodsong81) [inline mpa/cls/inferrer.py]: Double quotes, please :)

File: mpa/det/inferrer.py [hunk 3]
```diff
@@ -190,7 +186,24 @@ def dummy_dump_saliency_hook(model, input, out):
...
+            feature_vectors = [feature_vectors[i] for i in range(dataset.num_samples)]
+            saliency_maps = [saliency_maps[i] for i in range(dataset.num_samples)]
+            if not dataset.merged_results:
+                eval_predictions = dataset.merge(eval_predictions)
+            else:
+                eval_predictions = dataset.merged_results
+
+        assert len(eval_predictions) == len(feature_vectors) == len(saliency_maps), \
+               'Number of elements should be the same, however, number of outputs are ' \
```
Comment (goodsong81) [inline mpa/det/inferrer.py]: Same here.

**Review by goodsong81 — CHANGES_REQUESTED** [review]
Need minor fixes.

**Review by goodsong81 — CHANGES_REQUESTED** [review]
Could you resolve the conflict?
Sorry for late review, it's my bad.

**Comment by dongkwan-kim01** [comment]
@goodsong81 Thank you for noticing. Resolved conflict and tested in the otx side.

**Review by goodsong81 — APPROVED** [review]

**Pull request merged**